ReSharper.FSharp/src/FSharp.Common/src/Shim/TypeProviders/ExtensionTypingProviderShim.fs

```diff
@@ -114,5 +116,11 @@ type ExtensionTypingProviderShim(solution: ISolution, toolset: ISolutionToolset,
 
             $"{inProcessDump}\n\n{outOfProcessDump}"
 
+        member this.HasGenerativeTypeProviders(assemblies) =
+            // We can determine which projects contain generative provided types
+            // only from type providers hosted out-of-process
+            not (isConnectionAlive()) ||

```

Comment:
Why should it be `true` when the connection is not alive?

Reply:
in case out-of-process type-providers are disabled, then the general check should be equivalent to a simple `project.IsFSharp` check. You seem to be right, there should be isConnectionAlive () && ...